internal-dns/types/src/config.rs

```diff
@@ -410,6 +410,32 @@ impl DnsConfigBuilder {
         self.service_backend_zone(ServiceName::Mgd, &zone, mgd_port)
     }
 
+    /// Higher-level shorthand for adding a Nexus zone with both its internal
+    /// API service and its lockstep API service.
+    ///
+    /// # Errors
+    ///
+    /// This function fails only if the given zone has already been added to the
+    /// configuration.
+    pub fn host_zone_nexus(
+        &mut self,
+        zone_id: OmicronZoneUuid,
+        internal_address: SocketAddrV6,
+        lockstep_port: u16,
+    ) -> anyhow::Result<()> {
+        let zone = self.host_zone(zone_id, *internal_address.ip())?;
+        self.service_backend_zone(
+            ServiceName::Nexus,
```

Comment:
There seem to be a few places where external callers still use `service_backend_zone` with `Nexus` in internal-dns/resolver/src/resolver.rs and sled-agent/src/fakes/nexus.rs.

Reply:
The `internal-dns/resolver/src/resolver.rs` uses are testing the resolver with a Progenitor client, so that probably does not need to be replaced with the method that adds both Nexus and NexusLockstep. (You could replace the service with any other service and it would remain a valid test.)

The latter is used during setting up a fake Nexus for testing sled-agent calls into it. I'm inclined to leave this as-is until I determine whether we need to add the NexusLockstep API to this fake for RSS-related tests in what should be the 3rd PR of this stack.